### ISSUE TITLE
Execution reverted when calling `nucypher stake increase`

### ISSUE THREAD

**Issue by cygnusv** [nucypher/nucypher#2399]
**Describe the Bug**
`nucypher stake increase` is not working for some stakers. This command should fire 2 TXs: one for approving StakingEscrow to spend some amount and another for actually calling the `depositAndIncrease()` method in StakingEscrow.
It seems the first TX is going through correctly, but the second is failing during construction for some unknown reason.

BTW, the staker tried this command several times (since that 2nd TX never went through) and this caused the allowance to increase each time. We should detect in the allowance is enough before increasing it.

**To Reproduce**
TBD

**Traceback or Screenshots (Optional) **

```
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 541, in build_contract_transaction
    logger=self.log)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 463, in _handle_failed_transaction
    raise transaction_failed from exception
```
`Reason: execution reverted`

**System (please complete the following information):**
 - OS Version:
 - Nucypher Version: 4.1.0
 - Ethereum Node Version:

**Comment by nickhill** [comment]
The initial error, first attempt: 
═════════════════════════════════════════════════════════════════════════
Confirm increase stake (index: 0) by 15000 NU? [y/N]: y
Enter password to unlock account <staker-address>: 
Broadcasting INCREASEALLOWANCE Transaction (0.001257750788597667 ETH @ 27.825729267 gwei)...
execution reverted
Traceback (most recent call last):
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 534, in build_contract_transaction
    transaction_dict = contract_function.buildTransaction(payload)  # Gas estimation occurs here
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/contract.py", line 1083, in buildTransaction
    **self.kwargs
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/contract.py", line 1647, in build_transaction_for_function
    prepared_transaction = fill_transaction_defaults(web3, prepared_transaction)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/_utils/transactions.py", line 87, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/_utils/transactions.py", line 55, in <lambda>
    'gas': lambda web3, tx: web3.eth.estimateGas(tx),
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/eth.py", line 450, in estimateGas
    params,
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/web3/manager.py", line 153, in request_blocking
    raise ValueError(response["error"])
ValueError: {'code': -32000, 'message': 'execution reverted'}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/nucypher2/.virtualenvs/phase4/bin/nucypher", line 8, in <module>
    sys.exit(nucypher_cli())
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  [Previous line repeated 1 more time]
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/cli/commands/stake.py", line 653, in increase
    receipt = STAKEHOLDER.increase_stake(stake=current_stake, amount=value, from_unlocked=from_unlocked)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/decorators.py", line 116, in wrapped
    return func(actor, *args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/actors.py", line 1051, in increase_stake
    receipt = self._deposit_and_increase(stake_index=stake.index, amount=int(amount))
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/actors.py", line 1162, in _deposit_and_increase
    amount=amount)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/agents.py", line 433, in deposit_and_increase
    sender_address=staker_address)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 674, in send_transaction
    use_pending_nonce=use_pending_nonce)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 541, in build_contract_transaction
    logger=self.log)
  File "/home/nucypher2/.virtualenvs/phase4/lib/python3.6/site-packages/nucypher/blockchain/eth/interfaces.py", line 463, in _handle_failed_transaction
    raise transaction_failed from exception
nucypher.blockchain.eth.interfaces.TransactionFailed: DEPOSITANDINCREASE from <staker-address>... 
Sender balance: <enough> ETH 
Reason: execution reverted 
Transaction: {'chainId': 1, 'nonce': 9, 'from': '<staker-address>'}

**Comment by nickhill** [comment]
Second transaction, different system than the first, wondered if older Python version caused prior problem:
═════════════════════════════════════════════════════════════════════════
Confirm increase stake (index: 0) by 15000 NU? [y/N]: y
Enter password to unlock account staker-address: 
Broadcasting INCREASEALLOWANCE Transaction (0.000813961113556521 ETH @ 26.951462321 gwei)...
Traceback (most recent call last):
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/eth.py", line 330, in waitForTransactionReceipt
    return wait_for_transaction_receipt(self.web3, transaction_hash, timeout, poll_latency)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/transactions.py", line 111, in wait_for_transaction_receipt
    _timeout.sleep(poll_latency)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/threads.py", line 88, in sleep
    self.check()
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/threads.py", line 81, in check
    raise self
web3._utils.threads.Timeout: 600 seconds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nucypher/.virtualenvs/nucypher/bin/nucypher", line 8, in <module>
    sys.exit(nucypher_cli())
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  [Previous line repeated 1 more time]
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/commands/stake.py", line 653, in increase
    receipt = STAKEHOLDER.increase_stake(stake=current_stake, amount=value, from_unlocked=from_unlocked)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 116, in wrapped
    return func(actor, *args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/actors.py", line 1051, in increase_stake
    receipt = self._deposit_and_increase(stake_index=stake.index, amount=int(amount))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/actors.py", line 1157, in _deposit_and_increase
    self.token_agent.increase_allowance(increase=amount,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/agents.py", line 181, in increase_allowance
    receipt: TxReceipt = self.blockchain.send_transaction(contract_function=contract_function,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/interfaces.py", line 682, in send_transaction
    txhash_or_receipt = self.sign_and_broadcast_transaction(transaction_dict=transaction,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/interfaces.py", line 617, in sign_and_broadcast_transaction
    receipt = self.client.wait_for_receipt(txhash, timeout=self.TIMEOUT, confirmations=confirmations)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/clients.py", line 330, in wait_for_receipt
    receipt = self.w3.eth.waitForTransactionReceipt(transaction_hash=transaction_hash,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/eth.py", line 332, in waitForTransactionReceipt
    raise TimeExhausted(
web3.exceptions.TimeExhausted: Transaction (second transaction) is not in the chain, after 600 seconds

**Comment by nickhill** [comment]
third attempt, same system as 2nd attempt
═════════════════════════════════════════════════════════════════════════
Confirm increase stake (index: 0) by 15000 NU? [y/N]: y
Enter password to unlock account (staker-address): 
Broadcasting INCREASEALLOWANCE Transaction (0.001093038048262842 ETH @ 36.192114442 gwei)...
Traceback (most recent call last):
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/eth.py", line 330, in waitForTransactionReceipt
    return wait_for_transaction_receipt(self.web3, transaction_hash, timeout, poll_latency)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/transactions.py", line 111, in wait_for_transaction_receipt
    _timeout.sleep(poll_latency)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/threads.py", line 88, in sleep
    self.check()
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/_utils/threads.py", line 81, in check
    raise self
web3._utils.threads.Timeout: 600 seconds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nucypher/.virtualenvs/nucypher/bin/nucypher", line 8, in <module>
    sys.exit(nucypher_cli())
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/options.py", line 163, in wrapper
    return func(**kwargs)
  [Previous line repeated 1 more time]
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/cli/commands/stake.py", line 653, in increase
    receipt = STAKEHOLDER.increase_stake(stake=current_stake, amount=value, from_unlocked=from_unlocked)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 116, in wrapped
    return func(actor, *args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/actors.py", line 1051, in increase_stake
    receipt = self._deposit_and_increase(stake_index=stake.index, amount=int(amount))
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/actors.py", line 1157, in _deposit_and_increase
    self.token_agent.increase_allowance(increase=amount,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/agents.py", line 181, in increase_allowance
    receipt: TxReceipt = self.blockchain.send_transaction(contract_function=contract_function,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/interfaces.py", line 682, in send_transaction
    txhash_or_receipt = self.sign_and_broadcast_transaction(transaction_dict=transaction,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/interfaces.py", line 617, in sign_and_broadcast_transaction
    receipt = self.client.wait_for_receipt(txhash, timeout=self.TIMEOUT, confirmations=confirmations)
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/nucypher/blockchain/eth/clients.py", line 330, in wait_for_receipt
    receipt = self.w3.eth.waitForTransactionReceipt(transaction_hash=transaction_hash,
  File "/home/nucypher/.virtualenvs/nucypher/lib/python3.8/site-packages/web3/eth.py", line 332, in waitForTransactionReceipt
    raise TimeExhausted(
web3.exceptions.TimeExhausted: Transaction (third transaction) is not in the chain, after 600 seconds

**Comment by koos34** [comment]
This sounds related to an issue I had, I tried using the stake-nucypher UI to add CASI tokens and it only completed the first approval transaction. I had to manually call the depositAndIncrease method in the staking escrow contract to complete the process. This successfully increased the stake.

**Comment by nickhill** [comment]
I don't see depositAndIncrease in the ABI for the staking escrow contract. How does one do that manually?

**Comment by koos34** [comment]
Using a browser and Metamask, yes. You can use the Etherscan interface for the proxy contract here: https://etherscan.io/address/0xbbD3C0C794F40c4f993B03F65343aCC6fcfCb2e2#writeProxyContract

Click ”Connect to Web3” to connect Metamask (link with a red dot on the left, at the top of the method list) and look for method 5. depositAndIncrease.

Arguments are 0 and the stake in nubits (15000000000000000000000 for a 15K nu stake).
Eg
	_index
		0
	_value
		15000000000000000000000
Click Write, if you don’t get any errors then approve the transaction via Metamask.

(Credit and thanks to Roma K for pointing this out).

**Comment by nickhill** [comment]
thanks koos34. I just saw your post above. I did attempt to do that very operation, but screwed up the NUbits. 3 times. I still don't have all my CASI NU added to my stake, but I wanted to see what happened with what I had already done.  Also, my stake has ceased compounding. I'm unfamiliar with the escrow contract logic still, and sideways enough that I'm stopping until I either get myself straight or have some guidance or preferrably both.

**Comment by KevinVitale** [comment]
I received this error tonight. I used the following command, with an explicit gas price:

`$ nucypher stake increase --gas-price 15 --network mainnet`

After the failure, I reran the command, without specifying a gas price.

Here is the traceback:
```
Broadcasting INCREASEALLOWANCE Transaction (0.000678195 ETH @ 15 gwei)...
execution reverted
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/interfaces.py", line 538, in build_contract_transaction
    transaction_dict = contract_function.buildTransaction(payload)  # Gas estimation occurs here
  File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1083, in buildTransaction
    **self.kwargs
  File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1647, in build_transaction_for_function
    prepared_transaction = fill_transaction_defaults(web3, prepared_transaction)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/usr/local/lib/python3.7/site-packages/web3/_utils/transactions.py", line 87, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "/usr/local/lib/python3.7/site-packages/web3/_utils/transactions.py", line 55, in <lambda>
    'gas': lambda web3, tx: web3.eth.estimateGas(tx),
  File "/usr/local/lib/python3.7/site-packages/web3/eth.py", line 450, in estimateGas
    params,
  File "/usr/local/lib/python3.7/site-packages/web3/manager.py", line 153, in request_blocking
    raise ValueError(response["error"])
ValueError: {'code': -32000, 'message': 'execution reverted'}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/nucypher", line 11, in <module>
    sys.exit(nucypher_cli())
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/cli/options.py", line 165, in wrapper
    return func(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/cli/options.py", line 165, in wrapper
    return func(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/cli/options.py", line 165, in wrapper
    return func(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/cli/commands/stake.py", line 660, in increase
    receipt = STAKEHOLDER.increase_stake(stake=current_stake, amount=value, from_unlocked=from_unlocked)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/decorators.py", line 116, in wrapped
    return func(actor, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/actors.py", line 1052, in increase_stake
    receipt = self._deposit_and_increase(stake_index=stake.index, amount=int(amount))
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/actors.py", line 1163, in _deposit_and_increase
    amount=amount)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/agents.py", line 433, in deposit_and_increase
    sender_address=staker_address)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/interfaces.py", line 678, in send_transaction
    use_pending_nonce=use_pending_nonce)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/decorators.py", line 105, in wrapped
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/interfaces.py", line 545, in build_contract_transaction
    logger=self.log)
  File "/usr/local/lib/python3.7/site-packages/nucypher/blockchain/eth/interfaces.py", line 467, in _handle_failed_transaction
    raise transaction_failed from exception
nucypher.blockchain.eth.interfaces.TransactionFailed: DEPOSITANDINCREASE from 0x... 
[...]
Reason: execution reverted 
Transaction: {'chainId': 1, 'nonce': 25, 'from': '0x...'}
```

**Comment by KPrasch** [comment]
Closed by #2867
